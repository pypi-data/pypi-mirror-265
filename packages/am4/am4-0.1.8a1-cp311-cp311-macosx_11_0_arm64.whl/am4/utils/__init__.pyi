from __future__ import annotations
from . import aircraft
from . import airport
from . import db
from . import demand
from . import game
from . import log
from . import route
from . import ticket
__all__ = ['aircraft', 'airport', 'db', 'demand', 'game', 'log', 'route', 'ticket']
__version__: str = '0.1.8'
