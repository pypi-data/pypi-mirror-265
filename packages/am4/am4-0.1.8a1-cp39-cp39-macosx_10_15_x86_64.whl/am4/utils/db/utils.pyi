from __future__ import annotations
__all__ = ['jaro_distance', 'jaro_winkler_distance']
def jaro_distance(a: str, b: str) -> float:
    ...
def jaro_winkler_distance(a: str, b: str) -> float:
    ...
