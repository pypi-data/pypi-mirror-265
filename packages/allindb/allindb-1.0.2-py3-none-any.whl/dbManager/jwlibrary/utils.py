from inspect import getmembers, isfunction

def askFor(question, validator, error=lambda: None, required=True, final=None):
    out = input(question)
    if required:
        while not validator(out):
            error()
            out = input(question)
    if not validator(out):
        error()
        out = final or input(question)
    return out


def askForSec(*args):
    try:
        return askFor(*args)
    except KeyboardInterrupt:
        raise KeyboardInterrupt()
    except:
        return askForSec(*args)


def imports(module):
    for val in getmembers(module, isfunction):
        yield val
