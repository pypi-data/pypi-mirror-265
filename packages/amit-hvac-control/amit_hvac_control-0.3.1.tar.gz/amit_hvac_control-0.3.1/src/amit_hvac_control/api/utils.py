from aiohttp import MultipartWriter


def get_multipart_data(post: dict):
    with MultipartWriter("form-data") as mp:
        for key, value in post.items():
            part = mp.append(str(value))

            del part._headers["Content-Type"]
            del part._headers["Content-Length"]
            part.set_content_disposition("form-data", name=key)
        return mp
