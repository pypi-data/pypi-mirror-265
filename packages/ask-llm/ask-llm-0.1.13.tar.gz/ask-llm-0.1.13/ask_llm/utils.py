import re

from contextlib import suppress
from inspect import getfullargspec
from typing import Any, Callable, Iterable

from jinja2 import Environment, BaseLoader
from langchain.output_parsers import PydanticOutputParser

from .schemas import BaseAnswer


def format_attributes(
    func: Callable, func_args: Iterable, context_dict: dict, raw_template: str
) -> tuple[str, dict]:
    """
    Formats the attributes of the method's parent class / obj.
    """
    varnames = func.__code__.co_varnames
    if "self" in varnames or "cls" in varnames:
        _reference = "self" if "self" in varnames else "cls"
        _object = func_args[0]
        placeholders = re.findall(r"\{\{(.*?)\}\}|\{%\s*(.*?)\s*%\}", raw_template)
        placeholders = [
            item.strip()
            for sublist in placeholders
            for item in sublist
            if item and _reference in item
        ]
        placeholders = list(set(placeholders))
        for placeholder in placeholders:
            _attribute = placeholder.split(f"{_reference}.")[-1]
            if "." in _attribute:
                attr_template_name = _attribute.replace(".", "__")
            else:
                attr_template_name = _attribute
            context_dict[attr_template_name] = get_nested_attr(_object, _attribute)
            raw_template = raw_template.replace(
                f"{_reference}.{_attribute}", attr_template_name
            )
    return raw_template, context_dict


def format_args(func: Callable, func_args: Iterable, context_dict: dict) -> dict:
    """
    Formats the arguments of a function.
    """
    if len(func_args) > 0:
        args_names = getfullargspec(func).args
        self_is_here = "self" in args_names
        if self_is_here:
            args_names = args_names[1:]
        for i, arg_name in enumerate(args_names):
            with suppress(IndexError):
                context_dict[arg_name] = func_args[i + 1 if self_is_here else i]
    return context_dict


def format_func_body_result(
    result: Any, raw_template: str, context_dict: dict
) -> tuple[str, dict]:
    """
    Formats the result of a function to be used in a prompt.
    """
    if result and "__result__" in raw_template:
        context_dict["result_"] = result
        raw_template = raw_template.replace("__result__", "result_")
    return raw_template, context_dict


def format_return_type_instructions(schema: Any, raw_template: str) -> str:
    """
    Returns the format instructions of a Pydantic schema.
    """
    if schema and issubclass(schema, BaseAnswer):
        raw_template += "\n\n" + output_parser(schema).get_format_instructions()
    return raw_template


def get_nested_attr(obj: Any, attr: str) -> Any:
    """
    Recusrively gets a nested attribute from an object.
    """
    for attr in attr.split("."):
        if hasattr(obj, attr) and not callable(getattr(obj, attr)):
            obj = getattr(obj, attr)
    return obj


def output_parser(schema: BaseAnswer) -> PydanticOutputParser:
    """
    Converts a Pydantic scheme to a LCEL compatible output parser.
    Can be used to either get format instructions in a prompt or to parse the llm output.
    """
    return PydanticOutputParser(pydantic_object=schema)


def clean_docstring(docstring: str) -> str:
    """
    Cleans a docstring from unnecessary spaces and new lines.
    """
    docstring = docstring.strip()
    docstring = re.sub(" +", " ", docstring)
    docstring = re.sub("\n+", "\n", docstring)
    return docstring


def render_template(context_dict: dict, raw_template: str) -> str:
    """
    Render the raw template with Jinja
    """
    environment = Environment(loader=BaseLoader())
    template = environment.from_string(raw_template)
    return template.render(**context_dict)
