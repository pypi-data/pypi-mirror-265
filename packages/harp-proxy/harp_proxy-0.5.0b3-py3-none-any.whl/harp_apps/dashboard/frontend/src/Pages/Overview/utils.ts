export const mapGetValues = (data: { value: number }[]) => data.map((d: { value: number }) => d.value)
