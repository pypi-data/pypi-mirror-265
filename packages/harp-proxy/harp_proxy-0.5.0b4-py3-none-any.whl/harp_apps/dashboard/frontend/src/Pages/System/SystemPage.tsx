import { Page, PageTitle } from "Components/Page"
import { Tab } from "ui/Components/Tabs"

import { SystemDependenciesTabPanel } from "./SystemDependenciesTabPanel.tsx"
import { SystemSettingsTabPanel } from "./SystemSettingsTabPanel.tsx"
import { SystemStorageTabPanel } from "./SystemStorageTabPanel.tsx"

export const SystemPage = () => {
  return (
    <Page title={<PageTitle title="System" description="Informations about the running instance." />}>
      <Tab.Group>
        <Tab.List as="nav" aria-label="Tabs">
          <Tab>Settings</Tab>
          <Tab>Storage</Tab>
          <Tab>Dependencies</Tab>
        </Tab.List>
        <Tab.Panels>
          <SystemSettingsTabPanel />
          <SystemStorageTabPanel />
          <SystemDependenciesTabPanel />
        </Tab.Panels>
      </Tab.Group>
    </Page>
  )
}
