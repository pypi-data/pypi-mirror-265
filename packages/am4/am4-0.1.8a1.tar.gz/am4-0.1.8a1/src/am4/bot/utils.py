import time
from typing import Literal

import discord
from am4.utils.aircraft import Aircraft
from am4.utils.airport import Airport
from am4.utils.demand import CargoDemand, PaxDemand
from am4.utils.game import User
from am4.utils.route import AircraftRoute
from am4.utils.ticket import CargoTicket, PaxTicket, VIPTicket
from discord import AllowedMentions
from discord.ext import commands

from ..common import (
    HELP_U_FOURX,
    HELP_U_INCOME_LOSS_TOL,
    HELP_U_LOAD,
)
from ..config import cfg
from ..db.client import pb
from ..db.user import UserExtra
from .errors import CustomErrHandler

GUIDE_DEV_ROLEID = 646148607636144131
STAR_ROLEID = 701410528853098497

COLOUR_GENERIC = discord.Colour(0x9FACBD)
COLOUR_ERROR = discord.Colour(0xCA7575)
COLOUR_WARNING = discord.Colour(0xFFA94F)
COLOUR_SUCCESS = discord.Colour(0x75CA83)
COLOUR_REALISM = discord.Colour(0x277ECD)
COLOUR_EASY = discord.Colour(0x1A7939)

IY = "<:economy:701335275896307742>"
IJ = "<:business:701335275669946431>"
IF = "<:first:701335275938381824>"
IL = "<:large:701335275690786817>"
IH = "<:heavy:701335275799969833>"
ICSV = "<:csv:1210823880172904448>"
IJSON = "<:json:1210823838334586880>"

HELP_TPD = (
    "**Number of departures per day, per aircraft**\n"
    "- if not provided or set to `AUTO`, the bot will attempt to maximise it.\n"
    "- if you provide a number, the bot will try to assign more aircraft to the same route to avoid wasting demand.\n"
    "  - to disable the cramming, append a `!` at the end (not recommended)."
)
HELP_CFG_ALG = (
    "**Configuration Algorithm**\n"
    f"- {','.join(f'`{c}`' for c in Aircraft.PaxConfig.Algorithm.__members__)} (pax)\n"
    f"- {','.join(f'`{c}`' for c in Aircraft.CargoConfig.Algorithm.__members__)} (cargo)\n"
    "The best algorithm is picked automatically depending on the route distance.\n"
    "`YJF` here denotes the order of priority when filling seats."
)
HELP_SETTING_KEY = (
    "**The setting key** - some important ones are:\n"
    f"- `fourx`: {HELP_U_FOURX}\n"
    f"- `income_loss_tol`: {HELP_U_INCOME_LOSS_TOL}\n"
    f"- `load`: {HELP_U_LOAD}\n"
)

_SP100 = " "
_SP050 = " "
_SP033 = " "
_SP025 = " "
_SP022 = " "
_SP016 = " "
_SPHAIR = " "
_SPPUNC = " "


async def fetch_user_info(ctx: commands.Context) -> tuple[User, UserExtra]:
    u = ctx.author
    r_role = discord.utils.get(u.roles, name="Realism")
    e_role = discord.utils.get(u.roles, name="Easy")
    gm_target = "Realism" if r_role is not None else "Easy"
    role_id = r_role.id if r_role is not None else e_role.id if e_role is not None else None

    user, user_extra, dbstatus = await pb.users.get_from_discord(u.name, u.display_name, gm_target.upper(), u.id)
    if dbstatus == "created":
        gm_reason = f" because of your <@&{role_id}> role" if role_id is not None else ""
        await ctx.send(
            embed=discord.Embed(
                title="Your account has been created.",
                description=(
                    f"Your game mode is now `{gm_target}`{gm_reason}.\n\n"
                    f"To view your settings, use `{cfg.bot.COMMAND_PREFIX}settings show`.\n"
                    f"Need help with settings? Try `{cfg.bot.COMMAND_PREFIX}help settings`."
                ),
                color=COLOUR_SUCCESS,
            ),
            allowed_mentions=AllowedMentions.none(),
        )
    if (r_role is not None and user.game_mode == User.GameMode.EASY) or (
        e_role is not None and user.game_mode == User.GameMode.REALISM
    ):
        gm_user = "Realism" if user.game_mode == User.GameMode.REALISM else "Easy"
        h = CustomErrHandler(ctx)
        embed = h._get_err_embed(
            title="Mismatched game mode!",
            description=(
                f"I detected the <@&{role_id}> role on your account, but your settings indicate "
                f"that you are in the `{gm_user}` game mode.\n"
            ),
            sugg_cmd_override=[f"{cfg.bot.COMMAND_PREFIX}settings set game_mode {gm_target}"],
        )
        await ctx.send(embed=embed, allowed_mentions=AllowedMentions.none())
    return user, user_extra


def get_user_colour(user: User) -> discord.Colour:
    return COLOUR_REALISM if user.game_mode == User.GameMode.REALISM else COLOUR_EASY


def format_flight_time(t: float, short: bool = False) -> str:
    d, hrs = divmod(t * 3600, 86400)
    fs = "%H:%M" if short else "%H:%M:%S"
    return (f"{int(d)}D" if d else "") + time.strftime(fs, time.gmtime(hrs))


def format_ap_short(ap: Airport, mode: Literal[0, 1, 2]) -> str:
    indicator = "┏" if mode == 0 else "┣" if mode == 1 else "┗"
    return f"`{indicator} {ap.iata}` {ap.name}, {ap.country}"


def format_demand(pax_demand: PaxDemand, as_cargo: bool = False) -> str:
    if as_cargo:
        cargo_demand = CargoDemand(pax_demand)
        return f"{IL}`{cargo_demand.l:<7}` {IH}`{cargo_demand.h:<7}`"
    return f"{IY}`{pax_demand.y:<4}` {IJ}`{pax_demand.j:<5}` {IF}`{pax_demand.f:<5}`"


def format_config(cfg: Aircraft.PaxConfig | Aircraft.CargoConfig) -> str:
    if isinstance(cfg, Aircraft.CargoConfig):
        return f"{IL}`{cfg.l:<7}` {IH}`{cfg.h:<7}`"
    return f"{IY}`{cfg.y:<4}` {IJ}`{cfg.j:<5}` {IF}`{cfg.f:<5}`"


def format_ticket(ticket: PaxTicket | CargoTicket | VIPTicket) -> str:
    if isinstance(ticket, CargoTicket):
        return f"{IL}`{ticket.l:<7.2f}` {IH}`{ticket.h:<7.2f}`"
    return f"{IY}`{ticket.y:<4}` {IJ}`{ticket.j:<5}` {IF}`{ticket.f:<5}`"


def format_modifiers(apr: Aircraft):
    return "".join(
        name
        for name, set_ in zip(
            ["+SPD", "-FUEL", "-CO2", "+4X"], [apr.speed_mod, apr.fuel_mod, apr.co2_mod, apr.fourx_mod]
        )
        if set_
    )


def format_warning(w: AircraftRoute.Warning):
    if w == AircraftRoute.Warning.ERR_NO_STOPOVER:
        return "No stopovers are possible"
    elif w == AircraftRoute.Warning.ERR_INSUFFICIENT_DEMAND:
        return "Insufficient demand"
    elif w == AircraftRoute.Warning.ERR_DISTANCE_TOO_SHORT:
        return "Distance too short"
    elif w == AircraftRoute.Warning.ERR_DISTANCE_TOO_LONG:
        return "Distance too long"
    elif w == AircraftRoute.Warning.ERR_RWY_TOO_SHORT:
        return "Runway too short"
    elif w == AircraftRoute.Warning.ERR_FLIGHT_TIME_ABOVE_SPECIFIED:
        return "Flight time above specified limit"
    elif w == AircraftRoute.Warning.REDUCED_CONTRIBUTION:
        return "Reduced contribution"
    elif w == AircraftRoute.Warning.ERR_DISTANCE_ABOVE_SPECIFIED:
        return "Distance above specified limit"
    elif w == AircraftRoute.Warning.ERR_TRIPS_PER_DAY_TOO_HIGH:
        return "Trips per day per aircraft is too high"
    else:
        return "Unknown reason"


def format_num(num: float, *_pos) -> str:
    num = float(f"{num:.3g}")
    magnitude = 0
    while abs(num) >= 1000:
        magnitude += 1
        num /= 1000.0
    return f"{num:.{magnitude}f}".rstrip("0").rstrip(".") + ["", "K", "M", "B", "T"][magnitude]
