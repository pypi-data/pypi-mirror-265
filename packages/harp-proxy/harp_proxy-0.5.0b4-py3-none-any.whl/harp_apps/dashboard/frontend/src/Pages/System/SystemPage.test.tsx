import { screen } from "@testing-library/react"
import { act } from "react-dom/test-utils"
import { expect, it, test, describe } from "vitest"

import { renderWithClient } from "tests/utils"

import { SystemPage } from "./SystemPage"

describe("SystemPage", () => {
  it("renders the title and data when the query is successful", async () => {
    const result = renderWithClient(<SystemPage />)

    await result.findByText("endpoint1")
    expect(result.container).toMatchSnapshot()
  })

  test("clicking on a tab changes the content", async () => {
    const result = renderWithClient(<SystemPage />)

    await result.findByText("endpoint1")

    const tab = screen.getByText("Settings")
    act(() => {
      tab.click()
    })

    await result.findByText("proxy")
    expect(result.findByText("description1")).toBeTruthy()
  })
})
