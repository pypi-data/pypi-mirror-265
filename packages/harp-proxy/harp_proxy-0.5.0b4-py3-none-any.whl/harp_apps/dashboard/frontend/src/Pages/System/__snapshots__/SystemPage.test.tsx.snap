// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SystemPage > renders the title and data when the query is successful 1`] = `
<div>
  <main>
    <div
      class="mt-4 flex"
    >
      <div
        class="flex flex-col"
      >
        <h1
          class="css-9lrcch"
        >
          System
        </h1>
        <p
          class="css-11bp3eh"
        >
          Informations about the running instance.
        </p>
      </div>
    </div>
    <div
      class="mt-4"
    >
      <div
        class="sm:-mx-6 lg:-mx-8"
      >
        <section
          class="inline-block min-w-full max-w-full py-2 align-middle sm:px-6 lg:px-8"
        >
          <nav
            aria-label="Tabs"
            aria-orientation="horizontal"
            class="css-tvjki7"
            role="tablist"
          >
            <button
              aria-controls="headlessui-tabs-panel-:r3:"
              aria-selected="true"
              class="focus:ring-0 focus:ring-offset-0 focus:outline-none"
              data-headlessui-state="selected"
              id="headlessui-tabs-tab-:r0:"
              role="tab"
              tabindex="0"
              type="button"
            >
              <span
                class="css-7lc9z"
              >
                Settings
              </span>
            </button>
            <button
              aria-controls="headlessui-tabs-panel-:r4:"
              aria-selected="false"
              class="focus:ring-0 focus:ring-offset-0 focus:outline-none"
              data-headlessui-state=""
              id="headlessui-tabs-tab-:r1:"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="css-gebh7q"
              >
                Storage
              </span>
            </button>
            <button
              aria-controls="headlessui-tabs-panel-:r5:"
              aria-selected="false"
              class="focus:ring-0 focus:ring-offset-0 focus:outline-none"
              data-headlessui-state=""
              id="headlessui-tabs-tab-:r2:"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="css-gebh7q"
              >
                Dependencies
              </span>
            </button>
          </nav>
          <div
            class="css-ii4d2e"
          >
            <div
              aria-labelledby="headlessui-tabs-tab-:r0:"
              class="css-hu3rmt"
              data-headlessui-state="selected"
              id="headlessui-tabs-panel-:r3:"
              role="tabpanel"
              tabindex="0"
            >
              <div
                class="css-o0j4j7"
              >
                <table
                  class="min-w-fit divide-y divide-gray-300"
                >
                  <tbody
                    class="divide-y divide-gray-200 bg-white"
                  >
                    <tr>
                      <td
                        class="whitespace-nowrap py-2 px-4 text-sm font-medium text-gray-900 min-w-fit align-top"
                      >
                        proxy
                      </td>
                      <td
                        class="whitespace-nowrap text-sm text-gray-500"
                      >
                        <table
                          class="min-w-fit divide-y divide-gray-300"
                        >
                          <tbody
                            class="divide-y divide-gray-200 bg-white"
                          >
                            <tr>
                              <td
                                class="whitespace-nowrap py-2 px-4 text-sm font-medium text-gray-900 min-w-fit align-top"
                              >
                                endpoints
                              </td>
                              <td
                                class="whitespace-nowrap text-sm text-gray-500"
                              >
                                <table
                                  class="min-w-fit divide-y divide-gray-300"
                                >
                                  <tbody
                                    class="divide-y divide-gray-200 bg-white"
                                  >
                                    <tr>
                                      <td
                                        class="whitespace-nowrap py-2 px-4 text-sm font-medium text-gray-900 min-w-fit align-top"
                                      >
                                        0
                                      </td>
                                      <td
                                        class="whitespace-nowrap text-sm text-gray-500"
                                      >
                                        <table
                                          class="min-w-fit divide-y divide-gray-300"
                                        >
                                          <tbody
                                            class="divide-y divide-gray-200 bg-white"
                                          >
                                            <tr>
                                              <td
                                                class="whitespace-nowrap py-2 px-4 text-sm font-medium text-gray-900 min-w-fit align-top"
                                              >
                                                name
                                              </td>
                                              <td
                                                class="whitespace-nowrap text-sm text-gray-500"
                                              >
                                                endpoint1
                                              </td>
                                            </tr>
                                            <tr>
                                              <td
                                                class="whitespace-nowrap py-2 px-4 text-sm font-medium text-gray-900 min-w-fit align-top"
                                              >
                                                port
                                              </td>
                                              <td
                                                class="whitespace-nowrap text-sm text-gray-500"
                                              >
                                                8080
                                              </td>
                                            </tr>
                                            <tr>
                                              <td
                                                class="whitespace-nowrap py-2 px-4 text-sm font-medium text-gray-900 min-w-fit align-top"
                                              >
                                                url
                                              </td>
                                              <td
                                                class="whitespace-nowrap text-sm text-gray-500"
                                              >
                                                http://localhost:8080
                                              </td>
                                            </tr>
                                            <tr>
                                              <td
                                                class="whitespace-nowrap py-2 px-4 text-sm font-medium text-gray-900 min-w-fit align-top"
                                              >
                                                description
                                              </td>
                                              <td
                                                class="whitespace-nowrap text-sm text-gray-500"
                                              >
                                                description1
                                              </td>
                                            </tr>
                                          </tbody>
                                        </table>
                                      </td>
                                    </tr>
                                    <tr>
                                      <td
                                        class="whitespace-nowrap py-2 px-4 text-sm font-medium text-gray-900 min-w-fit align-top"
                                      >
                                        1
                                      </td>
                                      <td
                                        class="whitespace-nowrap text-sm text-gray-500"
                                      >
                                        <table
                                          class="min-w-fit divide-y divide-gray-300"
                                        >
                                          <tbody
                                            class="divide-y divide-gray-200 bg-white"
                                          >
                                            <tr>
                                              <td
                                                class="whitespace-nowrap py-2 px-4 text-sm font-medium text-gray-900 min-w-fit align-top"
                                              >
                                                name
                                              </td>
                                              <td
                                                class="whitespace-nowrap text-sm text-gray-500"
                                              >
                                                endpoint2
                                              </td>
                                            </tr>
                                            <tr>
                                              <td
                                                class="whitespace-nowrap py-2 px-4 text-sm font-medium text-gray-900 min-w-fit align-top"
                                              >
                                                port
                                              </td>
                                              <td
                                                class="whitespace-nowrap text-sm text-gray-500"
                                              >
                                                8081
                                              </td>
                                            </tr>
                                            <tr>
                                              <td
                                                class="whitespace-nowrap py-2 px-4 text-sm font-medium text-gray-900 min-w-fit align-top"
                                              >
                                                url
                                              </td>
                                              <td
                                                class="whitespace-nowrap text-sm text-gray-500"
                                              >
                                                http://localhost:8081
                                              </td>
                                            </tr>
                                            <tr>
                                              <td
                                                class="whitespace-nowrap py-2 px-4 text-sm font-medium text-gray-900 min-w-fit align-top"
                                              >
                                                description
                                              </td>
                                              <td
                                                class="whitespace-nowrap text-sm text-gray-500"
                                              >
                                                description2
                                              </td>
                                            </tr>
                                          </tbody>
                                        </table>
                                      </td>
                                    </tr>
                                  </tbody>
                                </table>
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <span
              aria-hidden="true"
              aria-labelledby="headlessui-tabs-tab-:r1:"
              id="headlessui-tabs-panel-:r4:"
              role="tabpanel"
              style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
              tabindex="-1"
            />
            <span
              aria-hidden="true"
              aria-labelledby="headlessui-tabs-tab-:r2:"
              id="headlessui-tabs-panel-:r5:"
              role="tabpanel"
              style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
              tabindex="-1"
            />
          </div>
        </section>
      </div>
    </div>
  </main>
</div>
`;
